js/ui/date_box/ui.date_box.base.js

```diff
@@ -407,42 +399,45 @@ var DateBox = DropDownEditor.inherit({
         $element.addClass(DATEBOX_CLASS + "-" + this._pickerType);
     },
 
-    _updateSize: function() {
-        var $element = this.$element(),
-            widthOption = this.option("width"),
-            isWidthSet = typeUtils.isDefined(widthOption) || (isRealWidthSet($element) && !this._isSizeUpdatable),
-            pickerType = this._pickerType,
-            // NOTE: no calculateWidth if type is rollers, why?
-            shouldCalculateWidth = pickerType !== PICKER_TYPE.rollers && devices.current().platform === "generic";
-
-        if(!windowUtils.hasWindow() || isWidthSet || !(shouldCalculateWidth && $element.is(":visible"))) {
+    _formatValidationIcon: function() {
+        if(!windowUtils.hasWindow()) {
             return;
         }
 
         var format = this._strategy.getDisplayFormat(this.option("displayFormat")),
             longestValue = dateLocalization.format(uiDateUtils.getLongestDate(format, dateLocalization.getMonthNames(), dateLocalization.getDayNames()), format);
-
-        $element.width(this._calculateWidth(longestValue));
-        this._isSizeUpdatable = true;
-    },
-
-    _calculateWidth: function(value) {
-        var IE_ROUNDING_ERROR = 10;
-        var NATIVE_BUTTONS_WIDTH = 48;
         var $input = this._input();
-        var $longestValueElement = dom.createTextElementHiddenCopy($input, value);
+        var $editorInput = $input.get(0);
+        var $dateBox = this.$element();
+
+        if(this._needsValidationIconHiding(longestValue, $dateBox, $input, $editorInput)) {
```

Comment:
Is it show validation icon after update widget's width runtime?

```suggestion
        var shouldHideValidationIcon = this._needsValidationIconHiding(longestValue, $dateBox, $input, $editorInput); 
        
        $dateBox.toggleClass(DX_INVALID_BADGE_CLASS, !shouldHideValidationIcon );
        if(shouldHideValidationIcon ) {
```
and remove L416-L418

Reply:
No. But it's good test case, thanks. I've tried to repair. Please, check it.